js_hw_11/task_02.ts

```diff
@@ -0,0 +1,81 @@
+// Создайте дженерик класс Storage<T>, где T должен быть ограничен объектом, имеющим КАК МИНИМУМ {id: number}.
+// Задача класса - хранить объекты типа Т в приватном массиве
+// Реализуйте в классе следующие методы:
+//   - constructor должен принимать необзятельный массив объектов соответствующего типа. 
+//     Если массив пришел - объекты запушить в хранилище.
+//   - add, принимающий либо объект типа Т, либо объект типа Т без id. Метод должен быть реализовать с помощью ПЕРЕГРУЗКИ.
+//     Если на вход подан объект без айди - айди надо сгенерировать, затем запушить обьект в хранилище
+//     Если на вход подан объект с айди - запушить его в хранилище
+//     Для типизации используйте Utility Types
+//   - update, принимающий объект с айди и любым набором остальных ключей из типа Т. 
+//   - remove, принимающий на вход id и удаляющий объект из массива
+//   - getById(id), возвращающий объект по айди если найден
+//   - getAll(), возвращает все объекты в хранилище
+
+type TObj = {
+    id: number
+};
+
+class MyStorage<T extends TObj>{
+    private data: T[] = [];
+    constructor(arr?: T[]){
+        if (arr) {
+            this.data.push(...arr)
+        }
+    };
+    private getNewId(objArr:T[]):number {
```

Comment:
```suggestion
    private getNewId():number {
```
No need to have argument in this method

Reply:
Done